### PR TITLE
Fixes slice bounds out of range error

### PR DESCRIPTION
**Pull request by jazanne**
* update truncateLine to count runes
* truncateLine maxCharCount is configurable
* add test cases

### CODE REVIEW

File: search/search.go [hunk 1]
```diff
@@ -25,19 +26,13 @@ const (
 // Truncate lines to prevent sending over massive hunks, e.g. a minified file.
 // NOTE: We may end up truncating a valid flag key reference. We accept this risk
 //       and will handle hunks missing flag key references on the frontend.
-func truncateLine(line string) string {
-	// len(line) returns number of bytes, not num. characters, but it's a close enough
-	// approximation for our purposes
-	if len(line) <= maxLineCharCount {
+func truncateLine(line string, maxCharCount int) string {
```
Comment (jazanne) [inline search/search.go:29]: easier for testing, but also just a better design not to use a global constant within this function

File: search/search.go [hunk 2]
```diff
@@ -25,19 +26,13 @@ const (
...
+	if utf8.RuneCountInString(line) <= maxCharCount {
```
Comment (jazanne) [inline search/search.go:30]: turns out this is faster than len([]runes) and it reads a lot better too

File: search/search_test.go [hunk 3]
```diff
@@ -400,6 +400,46 @@ func Test_SearchForRefs(t *testing.T) {
...
+func Test_truncateLine(t *testing.T) {
+	tests := []struct {
+		name         string
+		line         string
+		maxCharCount int
+		want         string
+	}{
+		{
+			name: "line equal to max character count returns line",
+			// len(line) = 8
```
Comment (InTheCloudDan) [inline search/search_test.go:412]: Everything else looks good to me, can we just remove these commented lines if not needed?

**Review by InTheCloudDan — CHANGES_REQUESTED** [review]

File: search/search_test.go (hunk 3 above)
Comment (jazanne) [reply search/search_test.go:412]: i was keeping them in for reference about the diff between line length and rune length - what if i consolidate on one line?

**Pull request merged**